mongo/mongo.go

```diff
@@ -681,25 +686,21 @@ func logVersion(mongoPath string) {
 	logger.Debugf("using mongod: %s --version: %q", mongoPath, output)
 }
 
-func installMongod(mongoDep packaging.Dependency, hostSeries, dataDir string) error {
-	// If we are not forcing a mongo snap via a feature flag, install the
-	// package list (which may also include snaps for focal+) provided by
-	// the mongo dependency for our series.
-	if !featureflag.Enabled(feature.MongoDbSnap) {
-		return packaging.InstallDependency(mongoDep, hostSeries)
-	}
-
+func installMongod(mongoDep packaging.Dependency, usingMongoFromSnap bool, hostSeries, dataDir string) (err error) {
 	snapName := ServiceName
-	jujuDbLocalSnapPattern := regexp.MustCompile(`juju-db_[0-9]+\.snap`)
-
-	// If we're installing a local snap, then provide an absolute path
-	// as a snap <name>. snap install <name> will then do the Right Thing (TM).
-	files, err := ioutil.ReadDir(path.Join(dataDir, "snap"))
-	if err == nil {
-		for _, fullFileName := range files {
-			_, fileName := path.Split(fullFileName.Name())
-			if jujuDbLocalSnapPattern.MatchString(fileName) {
-				snapName = fullFileName.Name()
+
+	if usingMongoFromSnap {
+		jujuDbLocalSnapPattern := regexp.MustCompile(`juju-db_[0-9]+\.snap`)
+
+		// If we're installing a local snap, then provide an absolute path
+		// as a snap <name>. snap install <name> will then do the Right Thing (TM).
+		files, err := ioutil.ReadDir(path.Join(dataDir, "snap"))
+		if err == nil {
+			for _, fullFileName := range files {
+				_, fileName := path.Split(fullFileName.Name())
+				if jujuDbLocalSnapPattern.MatchString(fileName) {
+					snapName = fullFileName.Name()
+				}
 			}
 		}
 	}
```

Comment:
Presumably the below dependency on "core" will change with the new versions of mongo 4.4? (but I guess that would only be in develop)